localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -806,6 +848,23 @@ geometry_msgs::msg::PoseWithCovarianceStamped NDTScanMatcher::align_using_monte_
       particle, i);
     ndt_monte_carlo_initial_pose_marker_pub_->publish(marker_array);
 
+    const geometry_msgs::msg::Pose pose = matrix4f_to_pose(ndt_result.pose);
+    geometry_msgs::msg::Vector3 rpy = get_rpy(pose);
+
+    TreeStructuredParzenEstimator::Input result(is_loop_variable.size());
+    result[0] = (pose.position.x - initial_pose_with_cov.pose.pose.position.x) / stddev_x;
+    result[1] = (pose.position.y - initial_pose_with_cov.pose.pose.position.y) / stddev_y;
+    result[2] = (pose.position.z - initial_pose_with_cov.pose.pose.position.z) / stddev_z;
+    result[3] = (rpy.x - base_rpy.x) / stddev_roll;
+    result[4] = (rpy.y - base_rpy.y) / stddev_pitch;
+    result[5] = (rpy.z - base_rpy.z) / M_PI;
+    result[0] = normal_to_uniform(result[0]);
+    result[1] = normal_to_uniform(result[1]);
+    result[2] = normal_to_uniform(result[2]);
+    result[3] = normal_to_uniform(result[3]);
+    result[4] = normal_to_uniform(result[4]);
```

Comment:
Only `yaw(result[5])` is a loop variable, so the conversion method is different, right? 
Please add the reason in a comment.

Reply:
That is correct. I have added comments and some refactoring to make the intent clearer.
https://github.com/autowarefoundation/autoware.universe/pull/5078/commits/696785099bee16b844c0ec02ead8671a5f6d933c

Reply:
Thank you. It is now easier to understand. :clap: